### PR TITLE
Xuejun/remove api serialize

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - remove deprecated api Serialize(std::ostream& xmlFile, ...
 - remove deprecated api Serialize(const std::string& xmlPath,...
 - remove deprecated api StreamSerialize(std::ostream& stream....

### Tickets:
 - 100091

### CODE REVIEW

File: src/inference/src/cnn_network_ngraph_impl.cpp [hunk 1]
```diff
@@ -589,14 +584,9 @@ StatusCode CNNNetworkNGraphImpl::serialize(const std::string& xmlPath,
...
     try {
-        std::map<std::string, ngraph::OpSet> custom_opsets;
-        for (const auto& extension : _ie_extensions) {
-            auto opset = extension->getOpSets();
-            custom_opsets.insert(begin(opset), end(opset));
-        }
         ngraph::pass::Manager manager;
         using namespace ov::pass;
-        REGISTER_PASS(manager, Serialize, xmlBuf, binBuf, custom_opsets, ov::pass::Serialize::Version::IR_V10)
+        REGISTER_PASS(manager, Serialize, xmlBuf, binBuf, ov::pass::Serialize::Version::IR_V10)
```
Comment (ilya-lavrenov) [inline src/inference/src/cnn_network_ngraph_impl.cpp:589]: looks like we still use extensions here. 
I suppose a simple removal can break user's code?

Comment (ilyachur) [reply src/inference/src/cnn_network_ngraph_impl.cpp:589]: Looks like we will be able to remove this code only together with IE API

Comment (zhaixuejun1993) [reply src/inference/src/cnn_network_ngraph_impl.cpp:589]: @ilyachur & @ilya-lavrenov So for this interfaces, we need to delay to next release?

Comment (ilya-lavrenov) [reply src/inference/src/cnn_network_ngraph_impl.cpp:589]: if one public API depends on another, we need to remove dependency first, then dependent API

**Comment by zhaixuejun1993** [comment]
Will delay this to next release.

**Pull request closed without merging**